### PR TITLE
nixos/tailscale: allow to set `TS_PERMIT_CERT_UID` env variable

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Description of changes

This setting was introduced with Tailscale 1.22.0, see
https://github.com/tailscale/tailscale/releases/tag/v1.22.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/tailscale.nix [hunk 1]
```diff
@@ -21,6 +21,12 @@ in {
       description = ''The interface name for tunnel traffic. Use "userspace-networking" (beta) to not use TUN.'';
     };
 
+    permitCertUid = mkOption {
+      type = types.str;
+      default = "";
```
Comment (pennae) [inline nixos/modules/services/networking/tailscale.nix:26]: using the empty string as a "not set" value isn't that great, prefer something like this:
```suggestion
      type = types.nullOr types.str;
      default = null;
```
and emit the env var only if the setting is not null

Comment (LeSuisse) [reply nixos/modules/services/networking/tailscale.nix:26]: Fine by me.

For Tailscale it does not really change a thing https://github.com/tailscale/tailscale/blob/c591c9165379f37bf3c3d56e9105b2e24f9f99d5/ipn/ipnserver/server.go#L459-L471

Comment (pennae) [reply nixos/modules/services/networking/tailscale.nix]: it's more for consistency with other modules. "null means not set" is a good convention to have :)

File: nixos/modules/services/networking/tailscale.nix [hunk 2]
```diff
@@ -38,7 +44,9 @@ in {
       serviceConfig.Environment = [
         "PORT=${toString cfg.port}"
         ''"FLAGS=--tun ${lib.escapeShellArg cfg.interfaceName}"''
-      ];
+      ] ++ (lib.optionals cfg.permitCertUid != null [
```
Comment (pennae) [inline nixos/modules/services/networking/tailscale.nix:47]:
```suggestion
      ] ++ (lib.optionals (cfg.permitCertUid != null) [
```

**Review by danderson — APPROVED** [review]

**Pull request merged**